src/binder/inc/assembly.hpp

```diff
@@ -26,16 +26,18 @@
 #if !defined(DACCESS_COMPILE) && !defined(CROSSGEN_COMPILE)
 #include "clrprivbinderassemblyloadcontext.h"
 #endif // !defined(DACCESS_COMPILE) && !defined(CROSSGEN_COMPILE)
-
-STDAPI BinderAcquirePEImage(LPCTSTR   szAssemblyPath,
-                            PEImage **ppPEImage,
-                            PEImage **ppNativeImage,
-                            BOOL      fExplicitBindToNativeImage);
-
-STDAPI BinderAcquireImport(PEImage                  *pPEImage,
-                           IMDInternalImport       **pIMetaDataAssemblyImport,
-                           DWORD                    *pdwPAFlags,
-                           BOOL                     bNativeImage);
+#include "bundle.h"
+
+STDAPI BinderAcquirePEImage(LPCTSTR     szAssemblyPath,
+                            PEImage   **ppPEImage,
+                            PEImage   **ppNativeImage,
+                            BOOL        fExplicitBindToNativeImage,
+                            BundleLoc   bundleLoc);
```

Comment:
Can we pass this as `const &` - const to make it explicit and `&` since it's two 64bit values...

Reply:
I originally used the type `const BundleLoc &`.  But this method is declared as `extern C` API, and I think the behavior of C++ reference parameters is not well-defined on such functions.  

I could use `const BundleLoc *`, but it'll need nullptr checks in a few places. Also, passing by value is convenient in a few other methods due to the use of default parameters. So, I left it passed by value here too.

Reply:
I wonder if the `STDAPI` is still needed...
@jkotas - Two questions:
* Is there a reason the binder APIs are declared as if exported? (although they really are not)
* What is the guidance on using a bit more modern C++ - like `byref` params, `const` and so on... in CLR ?

Reply:
This is left-over from fusion. Fusion was compiled as a separate thing, with its own infrastructure. Fusion did not include any VM headers, and VM did not include any fusion implementation headers. The hand shake between VM and fusion was done via `extern "C"` methods and COM interfaces.

> What is the guidance on using a bit more modern C++

As long as all compilers that we support work with it, it is ok.